### PR TITLE
Add yaml with exo export

### PR DESCRIPTION
**Pull request by danielwolff1**
In this PR, I implemented a method to export the mesh in exo format while dumping the .yaml input file.

When using the new method `dump_with_exo_mesh()`, `cubitpy` will automatically attach the `XYZ GEOMETRY` section to the yaml input file as well as omit the lengthy topology information, which will instead be inferred from the nodesets within 4C when reading in the exo mesh.

To fully enable this feature, I need to know which problems people at the institute are working with, in order to enable the output of the respective "GEOMETRY" section. Currently supported `PROBLEMTYPE`s are 
- `Fluid`
- `Structure`

but more can be added if needed, by extending the `problem_to_geometry_dict` inside the `add_problem_geometry_section()` function.

### CODE REVIEW

**Comment by danielwolff1** [comment]
@m-frey this might be interesting for you since we talked about it yesterday :)

File: src/cubitpy/cubit_to_fourc_input.py [hunk 1]
```diff
@@ -69,28 +70,105 @@ def add_node_sets(cubit, exo, input_file):
...
+        The relative path (as seen from the yaml input file) to the exodus
+        file that contains the mesh.
+    """
+
+    # map the problem type to the keyword for the problem specific geometry
+    # section
+    problem_to_geometry_dict = {
+        "Structure": "STRUCTURE GEOMETRY",
+        "Fluid": "FLUID GEOMETRY",
+    }
```
[the remark this answers is not in the document]
Comment (danielwolff1) [reply src/cubitpy/cubit_to_fourc_input.py:127]: Not sure I understand your comment here, maybe we can have chat about this?

[the remark this answers is not in the document]
Comment (isteinbrecher) [reply src/cubitpy/cubit_to_fourc_input.py:131]:
```python
    sections = {data[0].get_four_c_section() for data in cubit.blocks}
    if not len(sections) == 1:
         error
    problem_key = sections.pop() + " GEOMETRY"
```

**Review by isteinbrecher — DISMISSED** [review]
Thanks for the additional effort @danielwolff1, this looks good to me! If we want we can add the functionality for multi field input files (e.g., FSI) in a follow up PR.

**Comment by danielwolff1** [comment]
> Thanks for the additional effort @danielwolff1, this looks good to me! If we want we can add the functionality for multi field input files (e.g., FSI) in a follow up PR.

I now adapted the implementation and implemented a small test case to support the export of multiple fields. However, in the unit test, I decided not to replicate the correct definition of the boundary conditions from the 3D FSI tutorial, as the additional statements for creating the BCs would just blow up the test method and not really add anything, since the creation of the geometry section is independent of the creation of the boundary sections of the input files. Moreover, the latter are already tested in other functions. 

If you are fine with my proposal, this can be merged, otherwise I can also still adapt the test accordingly.

**Review by isteinbrecher — APPROVED** [review]
Thanks @danielwolff1 for these changes, very nice that we can now use the new 4C input features with directly reading exo files.
